### ISSUE TITLE
S3 protocol

### ISSUE THREAD

**Issue by federkasten** [totakke/cavia#7]
It would be nice to have support for the S3 protocol using MinIO and AWS S3 as the backend.

**Comment by totakke** [comment]
S3 protocol was implemented by b93a3d9 and released in [0.6.2](https://clojars.org/cavia/versions/0.6.2).

`:protocol :s3` must be specified for using S3 protocol.

```clj
(defprofile prof
  {:resources [{:id :resource
                ;; virtual hosted style
                :url "https://bucket-name.s3.region.amazonaws.com/resource"
                ;; or path style (incl. MinIO)
                ;; :url "https://s3.region.amazonaws.com/bucket-name/resource"
                :sha1 "0123456789abcdef01234567890abcdef0123456"
                :protocol :s3
                :auth {:access-key-id "accesskey", :secret-access-key "secretkey"}]})
```

**Comment by federkasten** [comment]
Thank you! @totakke :tada:
